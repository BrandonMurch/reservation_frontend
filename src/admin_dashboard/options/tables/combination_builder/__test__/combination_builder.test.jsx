// Dependencies
import React from 'react';
import {
  render, screen, createEvent, fireEvent, waitFor,
} from '@testing-library/react';
import { create } from 'react-test-renderer';

// Components
import CombinationBuilder from '../index';
import DraggableItem from 'general_components/draggable_list/draggable_item';
import { mockFetch } from 'test_utils';

const table = { name: 'name', seats: 2 };

const itemProps = {
  item: { name: 'name', seats: 2 },
  index: 0,
  DisplayComponent: ({ item }) => <p>{item.name}</p>,
  onDrop: jest.fn(),
  setHovered: jest.fn(),
};

describe('<CombinationBuilder />', () => {
  const fetchSpy = jest.spyOn(global, 'fetch')
    .mockImplementation(() => mockFetch(200));
  beforeEach(() => {
    render(
      <>
        <DraggableItem {...itemProps} />
        <CombinationBuilder />
      </>,
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const tree = create(
      <CombinationBuilder />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should order items when dragged and dropped', async () => {
    const combinationBuilder = screen.getByLabelText(/droppable/i);
    const dropEvent = createEvent.drop(combinationBuilder);
    Object.assign(dropEvent, { dataTransfer: { getData: jest.fn(() => JSON.stringify(table)) } });
    fireEvent(combinationBuilder, dropEvent);
    await waitFor(() => (
      expect(fetchSpy)
        .toHaveBeenCalled()
    ));
  });
});
