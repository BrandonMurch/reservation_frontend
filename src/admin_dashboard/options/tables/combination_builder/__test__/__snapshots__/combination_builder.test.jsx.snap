// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CombinationBuilder /> should match snapshot 1`] = `
<div
  className="container"
>
  <h1>
    Drag tables to create a combination
  </h1>
  <div
    aria-label="droppable"
    className="droppableBox"
    onDragOver={[Function]}
    onDrop={[Function]}
  />
  <button
    onClick={[Function]}
    type="button"
  >
    Create
  </button>
  <button
    type="button"
  >
    Cancel
  </button>
</div>
`;
